### PR TITLE
feat: struct update syntax

### PR DESCRIPTION
**Pull request by enitrat**
This change is [Reviewable](https://reviewable.io/reviews/cairo-book/cairo-book/761)

### CODE REVIEW

File: listings/ch05-using-structs-to-structure-related-data/listing_update_syntax/src/lib.cairo [hunk 1]
```diff
@@ -0,0 +1,23 @@
+use core::byte_array;
```
Comment (TAdev0) [inline listings/ch05-using-structs-to-structure-related-data/listing_update_syntax/src/lib.cairo:1]: import to remove

File: listings/ch05-using-structs-to-structure-related-data/listing_without_update_syntax/src/lib.cairo [hunk 2]
```diff
@@ -0,0 +1,27 @@
+use core::byte_array;
+#[derive(Drop)]
+struct User {
+    active: bool,
+    username: ByteArray,
+    email: ByteArray,
+    sign_in_count: u64,
+}
+
+// ANCHOR: here
```
Comment (TAdev0) [inline listings/ch05-using-structs-to-structure-related-data/listing_without_update_syntax/src/lib.cairo:10]: why do you have 2 anchors named here in the same file, in both listings? i guess you want to only shown the second one

File: listings/ch05-using-structs-to-structure-related-data/listing_without_update_syntax/src/lib.cairo [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+fn main() {
+    // --snip--
+    // ANCHOR_END: here
+
+    let user1 = User {
+        email: "someone@example.com", username: "someusername123", active: true, sign_in_count: 1,
+    };
+    // ANCHOR: here
+
```
Comment (TAdev0) [inline listings/ch05-using-structs-to-structure-related-data/listing_without_update_syntax/src/lib.cairo:19]: extra line after the anchor should be removed i suppose

File: src/ch05-01-defining-and-instantiating-structs.md [hunk 4]
```diff
@@ -70,3 +70,60 @@ Because the parameter names and the struct field names are exactly the same in L
...
+<span class="filename">Filename: src/lib.cairo</span>
+
+```rust
+{{#rustdoc_include ../listings/ch05-using-structs-to-structure-related-data/listing_without_update_syntax/src/lib.cairo:here}}
+```
+
+{{#label without-update-syntax}}
+
+<span class="caption">Listing {{#ref without-update-syntax}}: Creating a new `User` instance using all but one of
+the values from `user1`</span>
```
Comment (TAdev0) [inline src/ch05-01-defining-and-instantiating-structs.md:93]: we generally end with a point when commenting snippets with a listing

File: src/ch05-01-defining-and-instantiating-structs.md [hunk 5]
```diff
@@ -70,3 +70,60 @@ Because the parameter names and the struct field names are exactly the same in L
...
+Note that the struct update syntax uses `=` like an assignment; this is because it moves the data,
+just as we saw in the ["Moving Values"][move]<!-- ignore --> section. In this example, we can no
+longer use `user1` as a whole after creating `user2` because the `ByteArray` in the
+`username` field of `user1` was moved into `user2`. If we had given `user2` new
+`ByteArray` values for both `email` and `username`, and thus only used the
+`active` and `sign_in_count` values from `user1`, then `user1` would still be
+valid after creating `user2`. Both `active` and `sign_in_count` are types that
+implement the `Copy` trait, so the behavior we discussed in the ["`Copy` Trait"][copy]<!-- ignore --> section would apply.
+
+[move]: ch04-01-what-is-ownership.md#moving-values
```
Comment (TAdev0) [inline src/ch05-01-defining-and-instantiating-structs.md:128]: nice!

**Review by TAdev0 — APPROVED** [review]
@enitrat  left minor formatting comments, crystal clear with the precisions about `user1` being potentially still valid depending on the types used / how they are passed , very nice!
just one question:
```
Note that the struct update syntax uses `=` like an assignment; this is because it moves the data,
just as we saw in the ["Moving Values"][move]<!-- ignore --> section.
```
does this mean the struct update syntaxe case is different than the non update syntaxe case? doing `username: user1.username,` doens't pass the value?

File: listings/ch05-using-structs-to-structure-related-data/listing_without_update_syntax/src/lib.cairo (hunk 2 above)
Comment (enitrat) [reply listings/ch05-using-structs-to-structure-related-data/listing_without_update_syntax/src/lib.cairo:10]: you can have multiple anchors to only display separated parts of the code

Comment (TAdev0) [reply listings/ch05-using-structs-to-structure-related-data/listing_without_update_syntax/src/lib.cairo:9]: Oh I see you want to show twice the user1 yeah ok

**Pull request merged**